File: Phind-CodeLlama-34B-v2/context_template_full/loop_construct_collapse_clause.f90
#ifndef T1
!T1:loop construct collapse clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, DIMENSION(360, 10) :: a, b, c
        INTEGER :: i, j, k

        !$acc parallel loop collapse(2)
        DO i = 1, 360
          DO j = 1, 10
            a(i, j) = i + j
            b(i, j) = i - j
          END DO
        END DO
        !$acc end parallel loop

        !$acc parallel loop collapse(2)
        DO i = 1, 360
          DO j = 1, 10
            c(i, j) = a(i, j) * b(i, j)
          END DO
        END DO
        !$acc end parallel loop

        DO i = 1, 360
          DO j = 1, 10
            IF (c(i, j) .ne. a(i, j) * b(i, j)) THEN
              errors = errors + 1
            END IF
          END DO
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: Phind-CodeLlama-34B-v2/context_template_full/loop_construct_collapse_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
/*T1:loop construct collapse clause,V:2.0-2.7*/
int test1(){
    int err = 0;
    srand(SEED);

    real_t * a = (real_t *)malloc(1024 * sizeof(real_t));
    real_t * b = (real_t *)malloc(1024 * sizeof(real_t));

    for (int x = 0; x < 1024; ++x) {
        a[x] = rand() / (real_t)(RAND_MAX / 10);
        b[x] = 0.0;
    }

    #pragma acc data copy(a[0:1024], b[0:1024])
    {
        #pragma acc parallel loop collapse(2)
        for (int x = 0; x < 32; ++x) {
            for (int y = 0; y < 32; ++y) {
                int idx = x * 32 + y;
                b[idx] = a[idx] * 2.0;
            }
        }
    }

    for (int x = 0; x < 1024; ++x) {
        if (fabs(b[x] - (a[x] * 2.0)) > PRECISION) {
            err = 1;
            break;
        }
    }

    free(a);
    free(b);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}